### DISCUSSION TITLE
How to select book chapters and jump to a page?

### DISCUSSION THREAD

**Discussion by HolgerShadow** [sumatrapdfreader/sumatrapdf#3273] [General]
Hello!

I would like to know if you can see chapters of a book, the only thing I found was the option to jump to a particular page but when I put the page number in the search option it does not go to the page, as it does.

**Comment by kjk** [comment]
Currently it's not possible.

**Comment by GitHubRulesOK** [comment]
@DoctrineGamer 
The texts in a PDF have no way to differentiate between "Chapters" there are not such tags in a standard PDF they need to be deliberately added as part of a document overview "Outline" (often called "bookmarks") so there are tools that given a page number and text can easily retrospectively add an outline as bookmarks and thus if you list those pages, with the text you want, in a file, they can be appended to a PDF but that's a complex editing process that needs a lot of code as found in PDF editors. One simple tool is PDFtk but there are others   

 <strike>like Jbookmarks which I seem to recollect was allowed for as a side load ?</strike> no that was CmdOpenWith**PdfDjvuBookmarker** so see if that tool is of value https://sourceforge.net/projects/djvubookmarker/ however its 10 years old and has high dependencies like Acrobat X and Net 2.0 However the Manual is a good illustrative how to guide
![image](https://user-images.githubusercontent.com/12754913/218518126-3c19b39f-da9a-45c0-9814-7ac7c8949205.png)

**Comment by HolgerShadow** [comment]
Thanks for the suggestions, how is it not allowed to access the chapters of the books but and the pages search, how do I jump to such a page? It has a page search engine but when I type the page where I want to go it doesn't work.

**Comment by GitHubRulesOK** [comment]
@DoctrineGamer 
 text in pdf is not guaranteed, to be what the pixels are, some text may not be (under the surface) one letter after another, however if you have plain text showing as chap... it will go from one chap to the next chap, if they are there as basic text characters.
 If you are asking can you search the bookmarks area that is not generally possible as they are a separate table of links not stored as page plain text contents, but a stored named list built runtime as required for mouse selection.

**Comment by HolgerShadow** [comment]
Now I understand, thank you very much for your help.
